File: scripts/taskopt_client.py
#!/usr/bin/env python
import sys
import rospy
from tsu200.srv import TaskOpt


def call_taskopt(opt, id_val):
    rospy.wait_for_service('task_opt')
    try:
        svc = rospy.ServiceProxy('task_opt', TaskOpt)
        resp = svc(opt, id_val)
        return resp
    except rospy.ServiceException as e:
        rospy.logerr('Service call failed: %s', e)
        return None


if __name__ == '__main__':
    rospy.init_node('taskopt_client')
    opt = sys.argv[1] if len(sys.argv) > 1 else 'noop'
    id_val = int(sys.argv[2]) if len(sys.argv) > 2 else 0
    resp = call_taskopt(opt, id_val)
    if resp:
        print('success:', resp.success)
        print('status_code:', resp.status_code)
        print('message:', resp.message)

File: src/taskopt_client.cpp
#include <ros/ros.h>
#include "tsu200/TaskOpt.h"
#include <cstdlib>

int main(int argc, char **argv) {
  ros::init(argc, argv, "taskopt_client_cpp");
  ros::NodeHandle nh;
  ros::ServiceClient client = nh.serviceClient<tsu200::TaskOpt>("task_opt");
  tsu200::TaskOpt srv;
  std::string opt = "noop";
  int id = 0;
  if (argc > 1) opt = argv[1];
  if (argc > 2) id = std::atoi(argv[2]);
  srv.request.opt = opt;
  srv.request.id = id;
  if (client.call(srv)) {
    ROS_INFO("success=%s, code=%d, msg=%s", srv.response.success?"true":"false", srv.response.status_code, srv.response.message.c_str());
  } else {
    ROS_ERROR("Failed to call service task_opt");
    return 1;
  }
  return 0;
}
